splicemachine/ml/management.py

```diff
@@ -1,33 +1,165 @@
+    """
+    Get a readable version of the Pipeline stage
+    (without the memory address)
+    :param pipeline_stage: the name of the stage to parse
+    """
+    if '_' in str(pipeline_stage):
+        return str(pipeline_stage).split('_')[0]
+    return str(pipeline_stage)
+
+
+def _get_stages(pipeline):
+    """
+    Extract the stages from a fit or unfit pipeline
+
+    :param pipeline: a fit or unfit Spark pipeline
+    :return: stages list
+    """
+    if hasattr(pipeline, 'stages') and isinstance(pipeline.stages, list):
+        return pipeline.stages  # fit pipeline
+    return pipeline.getStages()  # unfit pipeline
+
+
+def _parse_string_parameters(string_parameters):
+    """
+    Parse string rendering of extractParamMap
+    :param string_parameters:
+    :return:
+    """
+    parameters = {}
+    parsed_mapping = str(string_parameters).replace("{", "").replace(
+        "}", "").replace("\n", "").replace("\t", "").split(',')
+
+    for mapping in parsed_mapping:
+        param = mapping[mapping.index('-') + 1:].split(':')[0]
+        value = mapping[mapping.index('-') + 1:].split(':')[1]
+
+        parameters[param.strip()] = value.strip()
+    return parameters
+
+
+def _get_cols(transformer, get_input=True):
+    """
+    Get columns from a transformer
+    :param transformer: the transformer (fit or unfit)
+    :param get_input: whether or not to return the input columns
+    :return: a list of either input or output columns
+    """
+    col_type = 'Input' if get_input else 'Output'
+    if hasattr(transformer, 'get' + col_type + 'Col'):  # single transformer 1:1 (regular)
+        col_function = transformer.getInputCol if get_input else transformer.getOutputCol
+        return [
+            col_function()] if get_input else col_function()  # so we don't need to change code for
+        # a single transformer
+    elif hasattr(transformer, col_type + "Col"):  # single transformer 1:1 (vec)
+        return getattr(transformer, col_type + "Col")
+    elif get_input and hasattr(transformer,
+                               'get' + col_type + 'Cols'):  # multi ple transformer n:1 (regular)
+        return transformer.getInputCols()  # we can never have > 1 output column (goes to vector)
+    elif get_input and hasattr(transformer, col_type + 'Cols'):  # multiple transformer n:1 (vec)
+        return getattr(transformer, col_type + "Cols")
+    else:
+        print(
+            "Warning: Transformer " + str(
+                transformer) + " could not be parsed. If this is a model, this is expected.")
+
 
 class MLManager(MlflowClient):
     """
     A class for managing your MLFlow Runs/Experiments
     """
 
-    def __init__(self, _tracking_uri=get_pod_uri("mlflow", "5001")):
-        mlflow.set_tracking_uri(_tracking_uri)
+    ARTIFACT_INSERT_SQL = 'INSERT INTO ARTIFACTS (run_uuid, path, "binary") VALUES (?, ?, ?)'
+    ARTIFACT_RETRIEVAL_SQL = 'SELECT "binary" FROM ARTIFACTS WHERE name=\'{name}\' ' \
```

Comment:
I believe for performance you want this to be a parameterized statement so that the statement is cached.  select "binary" from artifacts where name = ? and run_uuid = ?